README.md

```diff
@@ -24,6 +24,57 @@ In order to limit cognitive overhead prefer:
 MVC + `actions` + `jobs` + `services` + `workers`
 
 - Actions receive & validate user input that create/update records, with any related business logic.
-- Job have a single purpose with a clear start and end, usually asynchronous and ideally idempotent. 
+- Job have a single purpose with a clear start and end, usually asynchronous and ideally idempotent.
 - Services encapsulate complex logic or external calls that can be used in multiple places, usually jobs and actions.
 - Workers are long running processes listening for events, usually triggering relevant job in response.
+
+
+## Front end structure, rules and guidelines
+To keep the code clean, consistent and well understood across all projects, we should follow the conventions, rules and guidelines below, in addition to the rules and conventions specified by the linting rules and generally those adopted by the language or framework/s.
+
+General conventions
+- directories should use camelCase multiword names
+- regular js modules should follow the same directory naming conventions for components, and should be named in camelCase
+- if the js module exports a class, the filename should be in PascalCase
+- avoid relative imports -- prefer absolute paths always. http://codeofrob.com/entries/stop-using-relative-paths-in-your-javascripts.html
+
+Component Naming Conventions:
+- always use multiword component names to avoid naming collisions or nondescriptive components
+ - you should always know exactly what it is referring to based on the name
+- use PascalCase to name all vue components
+ - e.g. `TodoItem`
+- child components that are tightly coupled with the parent should include the parent component in its name as a prefix - https://vuejs.org/v2/style-guide/#Tightly-coupled-component-names-strongly-recommended
+- component names should start with the highest (most general) words and end with the descriptive words - https://vuejs.org/v2/style-guide/#Order-of-words-in-component-names-strongly-recommended
+- component names should always prefer full words over abbreviations - https://vuejs.org/v2/style-guide/#Full-word-component-names-strongly-recommended
+- avoid nested levels of folders
+ - the components should be grouped well and easily discoverable by name, and forces developers to give descriptive names following the naming conventions
+
+Page Naming Conventions:
+- page components should be prefixed with the resource they refer to
+  - `OrganisationsTable`, `BriefGroupForm`
+- for pages that list content for a resource, suffix pages with `List`
```

Comment:
Maybe `Index` to match rails backend?

Reply:
Not sure if we want to match rails necessarily as the frontend can and probably should use its own conventions that make sense, and I think `List` is a clearer name than `Index`. Happy to vote on this one though.

Reply:
I think `List` is nicer tbh. `Index` is very backend-ish.